script.js

```diff
@@ -1,4 +1,569 @@
+  while (diceArray.length > 0) {
+    // get hold of the largest num in the array (which is also the last element since we sorted it in ascending order earlier)
+    var largestNum = diceArray.pop();
+    // On the first iteration, there is no need to concatenate; just reassign playerNum
+    if (playerNum == null) {
+      playerNum = largestNum;
+    } else {
+      // else if it's not the first iteration, concatenate to get the largest possible number
+      playerNum = concatenate2Numbers(playerNum, largestNum);
+    }
+  }
+  // Update respective player's playerNum
+  currRoundParticipants[currPlayer].diceNum = playerNum;
+  // Return generated player num to parent function
+  return playerNum;
+};
+
+var sortAnArrayByASpecifiedKey = function (anArray, key) {
+  // get the length of the given array
+  var length = anArray.length;
+
+  //Use bubble sort to sort a given array
+  // More info on bubble sort: https://www.geeksforgeeks.org/bubble-sort/
+  for (var i = 0; i < length - 1; i += 1) {
+    for (var j = 0; j < length - 1; j += 1) {
+      if (anArray[j][key] > anArray[j + 1][key]) {
+        var temp = anArray[j];
+        anArray[j] = anArray[j + 1];
+        anArray[j + 1] = temp;
+      }
+    }
+  }
+  return anArray;
+};
+
+/**
+ * Compute the winner between all players.
+ * Return an array that is ordered according to who has the largest score.
+ */
+var determineWinner = function () {
+  // Overall summary of the logic: We'll start by arbitrarily assigning the first player to be the roundWinner. Then, on each subsequent player's turn, we'll assess if their diceNum is larger than the roundWinner. If yes: re-assign roundWinner; if No: do nothing
+
+  // If we've not yet set a roundWinner before (as will always be the case for the first player's turn), arbitrarily assign current player as the round winner
+
+  if (roundWinner == null) {
+    roundWinner = currRoundParticipants[currPlayer];
+  }
+
+  // If there is a previous roundWinner, assess if currPlayer's diceNum is larger than roundWinner's. If yes, this player becomes the new roundWinner
+  else if (currRoundParticipants[currPlayer].diceNum > roundWinner.diceNum) {
+    roundWinner = currRoundParticipants[currPlayer];
+  }
+  // If it's a draw, or the the currPlayer's diceNum is not larger than the existing roundWinner, do nothing
+};
+
+/**
+ * Incr the winner's points by adding previus round's points
+ * @param {number} roundScore
+ */
+var addNumToRunningScore = function (roundScore) {
+  // Get hold of the current players score, and add the roundScore to it
+  currRoundParticipants[currPlayer].score += roundScore;
+};
+
+var createLeaderBoardOutput = function () {
+  // set the preamble and assign it to a variable that will eventually be returned from this function
+  var leaderBoardOutput = 'Leaderboard: ';
+
+  // As per the game instructions, feel free to display the leaderboard in any order.
```

Comment:
This is nice, might be cleaner and more idiomatic to use `/* */` (single star) multi-line comment syntax for inline multi-line comments. thanks for explaining!